### PR TITLE
Ftr: add comment for cluster

### PR DESCRIPTION
**Pull request by watermelo**
**What this PR does**:
add comment for cluster directory

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by watermelo** [comment]
#564

File: cluster/cluster.go [hunk 1]
```diff
@@ -21,7 +21,8 @@ import (
 	"github.com/apache/dubbo-go/protocol"
 )
 
-// Cluster ...
+// Cluster
+//Extension - Cluster
```
Comment (zouyx) [inline cluster/cluster.go:25]:
```suggestion
// Extension - Cluster
```

File: cluster/loadbalance/least_active.go [hunk 2]
```diff
@@ -28,7 +28,6 @@ import (
 )
 
 const (
-	// LeastActive ...
 	LeastActive = "leastactive"
```
Comment (zouyx) [inline cluster/loadbalance/least_active.go:31]: I think should add comment for this public var

File: cluster/cluster_impl/failsafe_cluster.go [hunk 3]
```diff
@@ -31,7 +31,9 @@ func init() {
 	extension.SetCluster(failsafe, NewFailsafeCluster)
 }
 
-// NewFailsafeCluster ...
+// NewFailsafeCluster returns failsafe cluster instance
+//
+// Failure of security, anomalies, directly ignored. Usually used to write audit logs and other operations.
```
Comment (AlexStocks) [inline cluster/cluster_impl/failsafe_cluster.go:36]: // NewFailsafeCluster returns an failsafe cluster instance.
//
// Failure of security, anomalies, directly ignored. Usually it is 
//  used to write audit logs and other operations.

File: cluster/cluster_impl/failover_cluster.go [hunk 4]
```diff
@@ -31,7 +31,11 @@ func init() {
 	extension.SetCluster(name, NewFailoverCluster)
 }
 
-// NewFailoverCluster ...
+// NewFailoverCluster returns failover cluster instance
+//
+// Failure automatically switch, when there is failure,
+// retry the other server (default). Usually used for read operations,
+// but retries can result in longer delays.
```
Comment (AlexStocks) [inline cluster/cluster_impl/failover_cluster.go:38]: // NewFailoverCluster returns a failover cluster instance.
//
// Failure automatically switch, when there is a failure,
// retry the other server (default). Usually used for read operations,
// but retries can result in longer delays.

File: cluster/cluster_impl/failfast_cluster.go [hunk 5]
```diff
@@ -31,7 +31,10 @@ func init() {
 	extension.SetCluster(failfast, NewFailFastCluster)
 }
 
-// NewFailFastCluster ...
+// NewFailFastCluster returns failfast cluster instance
+//
+// Fast failure, only made a call, failure immediately error. Usually used for non-idempotent write operations,
+// such as adding records.
```
Comment (AlexStocks) [inline cluster/cluster_impl/failfast_cluster.go:37]: // NewFailFastCluster returns a failfast cluster instance.
//
// Fast failure, only made a call, failure immediately error. Usually used for non-idempotent write operations,
// such as adding records.

File: cluster/cluster_impl/broadcast_cluster.go [hunk 6]
```diff
@@ -31,7 +31,10 @@ func init() {
 	extension.SetCluster(broadcast, NewBroadcastCluster)
 }
 
-// NewBroadcastCluster ...
+// NewBroadcastCluster returns broadcast cluster instance
+//
+// Calling all providers broadcast, one by one call, any error is reported.
```
[the remark this answers is not in the document]
Comment (zouyx) [reply cluster/cluster_impl/broadcast_cluster.go]: what's the meaning of `providers'` ?

**Pull request merged**